src/core/debugger/debugger_interface.h

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: Copyright 2022 yuzu Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <functional>
+#include <span>
+#include <vector>
+
+#include "common/common_types.h"
+
+namespace Kernel {
+class KThread;
+}
+
+namespace Core {
+
+enum class DebuggerAction {
+    /**
+     * Stop emulation as soon as possible.
+     */
+    Interrupt,
+
+    /**
+     * Resume emulation.
+     */
+    Continue,
+
+    /**
+     * Step the currently-active thread.
+     */
+    StepThread,
+
+    /**
+     * Shut down the emulator.
+     */
+    ShutdownEmulation,
+};
```

Comment:
Since these are one liners, we may prefer them to be as such for less scrolling:
```suggestion
enum class DebuggerAction {
    Interrupt, // Stop emulation as soon as possible.
    Continue, // Resume emulation.
    StepThread, // Step the currently-active thread.
    ShutdownEmulation, // Shut down the emulator.
};
```
(and clang format so they are aligned)

Reply:
Generally for proper docstrings, you'd want to use `//<` to start the comment rather than `//`